File: java/sample.java
import org.sqlite.Function;
import java.sql.*;

public class sample {
    public static void main(String[] args) throws Exception {
        Class.forName("org.sqlite.JDBC");
        Connection conn = DriverManager.getConnection("jdbc:sqlite::memory:");

        // 사용자 정의 upper 함수 등록
        Function.create(conn, "upper", new Function() {
            @Override
            protected void xFunc() throws SQLException {
                if (args() == 1 && value_text(0) != null) {
                    result(value_text(0).toUpperCase());
                } else {
                    result();
                }
            }
        });

        String result = null;

        long start = System.currentTimeMillis();
        for (int i = 0; i < 100_000; i++) {
            try (PreparedStatement stmt = conn.prepareStatement("SELECT upper('hello world') as uppercased")) {
                ResultSet rs = stmt.executeQuery();
                if (rs.next()) {
                    result = rs.getString("uppercased");
                }
                rs.close();
            }
        }

        long end = System.currentTimeMillis();
        System.out.println(result); // HELLO WORLD
        System.out.println("Execution time: " + (end - start) + " ms");

        conn.close();
    }
}

File: python/sample.py
import sqlite3
import time

conn = sqlite3.connect(":memory:") # Or specify a database file path

def toUpperCase(value):
    return value.upper()

conn.create_function("upper", 1, toUpperCase);

cursor = conn.cursor()
cursor.execute("SELECT upper('hello world') as uppercased;")
result = cursor.fetchone()[0]
start_ms = time.time() * 1000  # 시작 시간 (밀리초)
for _ in range(100000):
    cursor.execute("SELECT upper('hello world') as uppercased;")
    result = cursor.fetchone()[0]
end_ms = time.time() * 1000  # 종료 시간 (밀리초)
print(result) # Output: HELLO WORLD
print(f"Execution time: {end_ms - start_ms} ms")
conn.close()